packages/ripple-binary-codec/src/binary.ts

```diff
@@ -81,13 +106,19 @@ function serializeObject(object: JsonObject, opts: OptionObject = {}): Buffer {
  *
  * @param transaction Transaction to serialize
  * @param prefix Prefix bytes to put before the serialized object
+ * @param opts.definitions Custom rippled types to use instead of the default. Used for sidechains and amendments.
  * @returns A Buffer with the serialized object
  */
 function signingData(
   transaction: JsonObject,
   prefix: Buffer = HashPrefix.transactionSig,
+  opts: { definitions?: XrplDefinitions } = {},
```

Comment:
A note about doing this (I realize it's done in other parts of the code too): in Python, if you make a param equal to a mutable object, like a dict, then mutate the param, it uses that same (already-mutated) object for every other call to that function. That's why you often see paradigms like this:
```py
def f(param=None):
  if param is None:
    param = {}
  ...
```
Not sure if this is also true in JS, but if it is, it's probably be a good idea to avoid doing that to future-proof the code.

Reply:
That's a super unexpected behavior. Let me check if it is in JS as well.